packages/components/src/button/style.scss

```diff
@@ -292,6 +292,16 @@
 	}
 }
 
+.button-bare {
+	appearance: none;
+	background: transparent;
+	border: none;
+	font-size: inherit;
```

Comment:
I tried add a `<Button bare>` and `<Button bare href>` to a paragraph of Calypso UI text, to see how well it blends into the background:

<img width="692" alt="bare-button-test" src="https://user-images.githubusercontent.com/664258/92220024-a7c0e680-ee9b-11ea-81ea-1f7402465e3b.png">

The `href` links blends perfectly, but the bare button needs a few more overrides:
```css
font-weight: inherit;
color: inherit;
```
There might be other `font-*` and other properties that need to be set, too.

Reply:
I added the `font-weight` and `color` resets in https://github.com/Automattic/wp-calypso/pull/45386/commits/9b0bb97a6edf213fc18d9582325afaf117c2c6f3

But making the `button` text blend into the background in all cases turns out to be a rather impossible task. For example, I can style the paragraph with:
```css
text-decoration: line-through;
text-shadow: 1px 1px black;
```
and the result is:

<img width="691" alt="Screenshot 2020-09-04 at 11 04 42" src="https://user-images.githubusercontent.com/664258/92222072-b6f56380-ee9e-11ea-8426-8a00bd754c8e.png">

The `button` doesn't inherit these. There's almost infinite number of text-affecting CSS properties like this.

I'm not sure if "blending into text background" is even a valid goal for `<Button bare>`. In most cases, it will be used to just provide a stable starting point for additional styling (with `<Button bare className={ ... }/>`).